lib/components/DownloadsScreen/downloads_overview.dart

```diff
@@ -1,185 +1,148 @@
+import 'dart:async';
+
 import 'package:auto_size_text/auto_size_text.dart';
-import 'package:finamp/services/downloads_helper.dart';
 import 'package:flutter/material.dart';
-import 'package:flutter_downloader/flutter_downloader.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
 import 'package:get_it/get_it.dart';
 
-import '../../services/download_update_stream.dart';
-import '../error_snackbar.dart';
+import '../../models/finamp_models.dart';
+import '../../services/isar_downloads.dart';
+import '../global_snackbar.dart';
 
 const double downloadsOverviewCardLoadingHeight = 120;
 
-class DownloadsOverview extends StatefulWidget {
-  const DownloadsOverview({Key? key}) : super(key: key);
-
-  @override
-  State<DownloadsOverview> createState() => _DownloadsOverviewState();
-}
-
-class _DownloadsOverviewState extends State<DownloadsOverview> {
-  late Future<List<DownloadTask>?> _downloadsOverviewFuture;
-  final _downloadUpdateStream = GetIt.instance<DownloadUpdateStream>();
-  final _downloadsHelper = GetIt.instance<DownloadsHelper>();
-
-  Map<String, DownloadTaskStatus>? _downloadTaskStatuses;
-
-  final Map<DownloadTaskStatus, int> _downloadCount = {
-    DownloadTaskStatus.undefined: 0,
-    DownloadTaskStatus.enqueued: 0,
-    DownloadTaskStatus.running: 0,
-    DownloadTaskStatus.complete: 0,
-    DownloadTaskStatus.failed: 0,
-    DownloadTaskStatus.canceled: 0,
-    DownloadTaskStatus.paused: 0,
-  };
-
-  bool _initialCountDone = false;
+class DownloadsOverview extends StatelessWidget {
+  const DownloadsOverview({super.key});
 
   @override
-  void initState() {
-    super.initState();
-    _downloadsOverviewFuture = FlutterDownloader.loadTasks();
+  Widget build(BuildContext context) {
+    final isarDownloads = GetIt.instance<IsarDownloads>();
 
-    // Like in DownloadedIndicator, we use our own listener instead of a
-    // StreamBuilder to ensure that we capture all events.
-    _downloadUpdateStream.stream.listen((event) {
-      if (_downloadTaskStatuses != null &&
-          _downloadTaskStatuses!.containsKey(event.id) &&
-          _downloadTaskStatuses![event.id] != event.status) {
-        setState(() {
-          _downloadCount[_downloadTaskStatuses![event.id]!] =
-              _downloadCount[_downloadTaskStatuses![event.id]]! - 1;
-          _downloadCount[event.status] = _downloadCount[event.status]! + 1;
-          _downloadTaskStatuses![event.id] = event.status;
-        });
+    isarDownloads.updateDownloadCounts();
+    // TODO is this a good idea?
+    isarDownloads.restartDownloads();
+    Timer.periodic(const Duration(seconds: 6), (timer) {
+      if (context.mounted) {
+        isarDownloads.updateDownloadCounts();
+      } else {
+        timer.cancel();
```

Comment:
This timer controls the song/image counts and the syncing count.  It's extra slow because these are full recounts of the database, although isar seems to do them pretty fast anyway.  The download counts on the right are throttled separately to one update per second because updating at every change choked the UI.  I can bump it to 10 per second and it should be fine.  I can probably bump this a bit too if you care.

Reply:
We can go with 5s and 0.2s, that should be fast enough?